modules/navier_stokes/src/fvbcs/INSFVMassAdvectionOutflowBC.C

```diff
@@ -44,34 +44,23 @@ INSFVMassAdvectionOutflowBC::INSFVMassAdvectionOutflowBC(const InputParameters &
              "'--with-ad-indexing-type=global'");
 #endif
 
-  if (!_u_var)
-    paramError("u", "the u velocity must be an INSFVVelocityVariable.");
-
-  if (_dim >= 2 && !_v_var)
-    paramError("v",
-               "In two or more dimensions, the v velocity must be supplied and it must be an "
-               "INSFVVelocityVariable.");
-
-  if (_dim >= 3 && !params.isParamValid("w"))
-    paramError("w",
-               "In three-dimensions, the w velocity must be supplied and it must be an "
-               "INSFVVelocityVariable.");
+  if (_dim >= 2 && !_v)
+    mooseError(
+        "In two or more dimensions, the v velocity must be supplied using the 'v' parameter");
+  if (_dim >= 3 && !_w)
+    mooseError("In threedimensions, the w velocity must be supplied using the 'w' parameter");
 }
 
 ADReal
 INSFVMassAdvectionOutflowBC::computeQpResidual()
 {
-#ifdef MOOSE_GLOBAL_AD_INDEXING
-  ADRealVectorValue v(_u_var->getBoundaryFaceValue(*_face_info));
-  if (_v_var)
-    v(1) = _v_var->getBoundaryFaceValue(*_face_info);
-  if (_w_var)
-    v(2) = _w_var->getBoundaryFaceValue(*_face_info);
+  const auto boundary_face = singleSidedFaceArg();
```

Comment:
same?

Reply:
we didn't do this before. Now you want to add it now that I've removed the variables?

Reply:
seems like we should have the option to be consistently using skewness-correction everywhere

Reply:
you can open an issue for it. Feature addition/control is not the aim of this PR

Reply:
For sure, but we need to agree that we ll do skewness correction at the kernel level if we re removing access to the variable.

It makes sense to realize now that we re not doing skewness-correction in all the operators as expected

Reply:
#23438